src/core/internal/lifetime.d

```diff
@@ -89,37 +89,92 @@ Emplaces T.init.
 In contrast to `emplaceRef(chunk)`, there are no checks for disabled default
 constructors etc.
 +/
+pragma(inline, true)
 void emplaceInitializer(T)(scope ref T chunk) nothrow pure @trusted
     if (!is(T == const) && !is(T == immutable) && !is(T == inout))
 {
-    import core.internal.traits : hasElaborateAssign;
+    static if (is(T U == shared U))
+        alias Unshared = U;
+    else
+        alias Unshared = T;
+
+    emplaceInitializerImpl!Unshared(chunk);
+}
 
-    static if (!hasElaborateAssign!T && __traits(compiles, chunk = T.init))
+private template emplaceInitializerImpl(T) {
+    import core.internal.traits : hasElaborateAssign;
+    static if (__traits(isZeroInit, T))
     {
-        chunk = T.init;
+        void emplaceInitializerImpl(T)(scope ref T chunk) nothrow pure @trusted @nogc
+        {
+            import core.stdc.string : memset;
+            memset(cast(void*) &chunk, 0, T.sizeof);
+        }
     }
-    else static if (__traits(isZeroInit, T))
+    else static if (T.sizeof <= 16 && !hasElaborateAssign!T && __traits(compiles, (){ T chunk; chunk = T.init; }))
     {
-        static if (is(T U == shared U))
-            alias Unshared = U;
-        else
-            alias Unshared = T;
-
-        import core.stdc.string : memset;
-        memset(cast(Unshared*) &chunk, 0, T.sizeof);
+        void emplaceInitializerImpl(T)(scope ref T chunk) nothrow pure @trusted @nogc
+        {
+            chunk = T.init;
+        }
     }
     else
     {
-        // emplace T.init (an rvalue) without extra variable (and according destruction)
-        alias RawBytes = void[T.sizeof];
-
-        static union U
+        version (DMD) version = UseInitSymbolName;
+        version (LDC) version = UseInitSymbolName;
+        version (UseInitSymbolName)
         {
-            T dummy = T.init; // U.init corresponds to T.init
-            RawBytes data;
+            static if (is(T U : U[N], size_t N)) // if isStaticArray
+            {
+                // For static arrays there is no initializer symbol created. Instead, we emplace elements one-by-one.
+                void emplaceInitializerImpl(T)(scope ref T chunk) nothrow pure @trusted @nogc
+                {
+                    foreach (i; 0..N)
+                    {
+                        emplaceInitializer(chunk[i]);
+                    }
+                }
+            }
+            else
+            {
+                // Avoid stack allocation by hacking to get to the init symbol.
+                pragma(mangle, "_D" ~ T.mangleof[1..$] ~ "6__initZ")
```

Comment:
@ibuclaw Can we also generate GDC's `.init` symbol names this way or with a slight tweak?

Reply:
Yes, initializers follow the same naming convention.